snapcraft/tests/test_plugin_nodejs.py

```diff
@@ -58,10 +59,34 @@ class Options:
                 path.join(os.path.abspath('.'), 'parts', 'test-part', 'npm')),
             mock.call().pull()])
 
+    def test_build_local_sources_and_node_folder(self):
+        class Options:
+            source = '.'
+            node_packages = []
+            node_folder = 'subdir'
+
+        plugin = nodejs.NodePlugin('test-part', Options())
+
+        os.makedirs(plugin.sourcedir)
+        os.makedirs(os.path.join(plugin.sourcedir, plugin.options.node_folder))
+        open(os.path.join(plugin.sourcedir, plugin.options.node_folder,
+                          'package.json'), 'w').close()
+
+        plugin.build()
+
+        self.run_mock.assert_has_calls([
+            mock.call(['npm', 'install', '-g'], cwd=plugin.builddir)])
```

Comment:
Yeah you're on the right track here. You just need to add the expected path to the `package.json` after the `'-g'` argument in the call.